### PR TITLE
First pass of migrating mushrooms adpater v1->v2

### PR DESCRIPTION
**Pull request by shaneahern**
@xcellsior here's a first pass at migrating a v1->v2, was unclear on a few aspects of this but let's use this draft PR to review things and help me understand things in more detail.

### CODE REVIEW

File: projects/_template/index.js [hunk 1]
```diff
@@ -24,7 +24,7 @@
   module.exports = {
     name: 'Template Project', // project name
     token: null,              // null, or token symbol if project has a custom token
-    category: 'assets',       // allowed values as shown on DefiPulse: 'Derivatives', 'DEXes', 'Lending', 'Payments', 'Assets'
+    category: 'Assets',       // allowed values as shown on DefiPulse: 'Derivatives', 'DEXes', 'Lending', 'Payments', 'Assets'
```
Comment (shaneahern) [inline projects/_template/index.js:27]: just a housekeeping fix, noticed that this fails due to case-sensitivity for category type

File: v2/projects/mushrooms/index.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  return await getTokenHolderMap().holders;
+}
+
+module.exports = {
+    /* Project Metadata */
+    name: 'Mushrooms Finance', // project name
+    token: "MM",              // null, or token symbol if project has a custom token
+    category: 'Assets',       // allowed values as shown on DefiPulse: 'Derivatives', 'DEXes', 'Lending', 'Payments', 'Assets'
+    start: 1604160000,        // unix timestamp (utc 0) specifying when the project began, or where live data begins
+    tokenHolderMap: [
```
Comment (shaneahern) [inline v2/projects/mushrooms/index.js]: not sure here if we need a tokenHolderMap when we have a tvl function?

File: v2/projects/mushrooms/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      {
+        tokens: async () => await getTokens(),
+        holders: async () => await getHolders(),
+      },
+    ],
+    tvl,
```
Comment (shaneahern) [inline v2/projects/mushrooms/index.js:63]: seems like the ability to have a tvl function here in a v2 adapter means we can pretty much use the same code as we had in the v1 adapter?

File: projects/_template/index.js (hunk 1 above)
Comment (xcellsior) [reply projects/_template/index.js:27]: this is probably why "assets" was way more commonly an issue than the others....

File: v2/projects/mushrooms/index.js (hunk 2 above)
Comment (xcellsior) [reply v2/projects/mushrooms/index.js]: We do not need both - I recommend having only one or the other.

**Review by xcellsior — COMMENTED** [review]
I do not think the V2 json style works here as implemented because it requires a balance call to the vault contract that is non-standard. In reality, the ABI call to "balance" is actually getting the supply of the ERC-20 token. 
So, all of the tokens in vaults are considered TVL since the tokens are created thru deposit of other tokens.

File: v2/projects/mushrooms/index.js (hunk 2 above)
Comment (ruffrey) [reply v2/projects/mushrooms/index.js]: I believe the Indexer will choose one, defaulting to `tvl()` first.

**Review by xcellsior — COMMENTED** [review]
V2 folder is for Token Holder Maps. Otherwise looks good.

**Comment by shaneahern** [comment]
Determined that because we are continuing to use tvl that we can keep the mushrooms adapter code as is and just enable it in the indexer projects-tvl.js, so closing this and opening a PR to enable in the indexer project

**Pull request closed without merging**